In this exercise, you will practice basic `dplyr` (the relevant package within `tidyverse`) operations on a dataset of gene expression levels across human tissues (from the GTEx Project). The dataset (`dt`) has one row per gene, with columns:

-   `Name`: the Ensembl gene ID\
-   `Description`: the gene symbol
-   One column per tissue containing the median expression levels (in TPM units)

------------------------------------------------------------------------

### Start by loading the data into R:

Note the arguments, which specify that the first two lines of the file will be skipped, that the file contains a header line that should be used to form column names, and that the columns are tab (`\t`) separated.

```{r}
dt <- read_delim("GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz", skip = 2, col_names = TRUE, delim = "\t")
```

### Q1. How many genes are in the dataset, and how many tissues are represented?

```{r}

library(tidyverse)
```

```{r}

#Number of Genes
dim(dt)
  #There are 56,200 genes in the dataset

#Number of Tissues
colnames(dt)
  #There are 54 tissue types in this dataset (the other 2 columns are gene identifiers)
```

------------------------------------------------------------------------

### Q2. How many genes are expressed above a given threshold in each tissue?

A common threshold is 1 TPM to define whether a gene is "expressed."\
Compute, for each of the three tissues, `Liver`, `Testis`, and `Whole Blood`, how many genes are expressed above this threshold.\
Then identify which tissue has the **most** and which has the **fewest** expressed genes. See this paper for one possible explanation: Xia et al. 2021: [PMC7891839](https://pmc.ncbi.nlm.nih.gov/articles/PMC7891839/).

Do the results change if you set the threshold higher, say at 10 TPM?

```{r}

liver <- dt %>%
  select(Liver) %>%
  filter(Liver >= 1) %>%
  nrow()
liver
#There are 13,499 genes

testis <- dt %>%
  select(Testis) %>%
  filter(Testis >= 1) %>%
  nrow()
testis
#There are 25,064 genes

whole_blood <- dt %>%
  select(`Whole Blood`) %>%
  filter(`Whole Blood` >= 1) %>%
  nrow()
whole_blood
#There are 11,616 genes

#----changing threshold to 10----

liver10 <- dt %>%
  select(Liver) %>%
  filter(Liver >= 10) %>%
  nrow()
liver10
#There are 5,132 genes

testis10 <- dt %>%
  select(Testis) %>%
  filter(Testis >= 10) %>%
  nrow()
testis10
#There are 11,471 genes

whole_blood10 <- dt %>%
  select(`Whole Blood`) %>%
  filter(`Whole Blood` >= 10) %>%
  nrow()
whole_blood
#There are 4,482 genes

#Testis has the most expressed genes, and Whole Blood has the fewest expressed genes
#If I change the threshold to 10, the results remain the same

#The paper states that "the testis expresses the largest number of genes of any mammalian organ", so this definitely explains our results showing testis with the most expressed genes
```

------------------------------------------------------------------------

### Q3. What are the 20 most highly expressed genes in the liver?

What do you notice about many of the genes in this list? Can you propose a hypothesis to explain your observation?

```{r}

liver_top <- dt %>%
  arrange(desc(Liver)) %>%
  select(Name, Description, Liver)

liver_top20 <- dt %>%
  arrange(desc(Liver)) %>%
  select(Name, Description, Liver) %>%
  head(20)

liver_top20

# Most (13/20) of the genes in this list are mitochondrial genes. 
# Mitochondria produce ATP, which is vital for a cell to function, especially for an organ like the liver, therefore it would make sense for mitochondrial genes to be highly expressed
```

------------------------------------------------------------------------

### Q4. Compare the mean expression levels of mitochondrial versus nuclear-encoded genes in the Liver.

Genes encoded by the mitochondrial genome have the gene symbol prefix `MT-`. The function `str_starts()` from the `stringr` package in `tidyverse` tests whether a string (or a vector of strings) start with a given pattern, returning TRUE/FALSE (or a boolean vector) as output.

Use `str_starts()` to identify which genes are mitochondrially encoded and store this information in a new Boolean column named `is_mitochondrial`. Next, compute the mean and median expression of mitochondrial genes in the liver, as well as the mean and median expression of nuclear-encoded genes in the liver and compare them. Do mitochondrial genes exhibit higher or lower mean and median expression than nuclear genes? What does the difference between the mean and median tell you about the distribution of gene expression.

```{r}

mit_genes <- str_starts(liver_top$Description, "MT-")

liver_top <- liver_top %>%
  mutate(is_mitochondrial = mit_genes)

liver_top %>%
  group_by(is_mitochondrial) %>%
  summarize(Liver_mt_mean = mean(Liver, na.rm = TRUE) , Liver_med_mean = median(Liver, na.rm = TRUE))

#The mean expression of mitochondrial genes in the liver is 11592.72
#The median expression of mitochondrial genes in the liver is 3.40258
#The mean expression of nuclear-encoded genes in the liver is 8.373839
#The median expression of nuclear-encoded genes in the liver is 0.00000

# Mitochondrial genes exhibit higher mean and median expression values than nuclear genes
# Looking at mitochondrial genes, the mean is so much higher than the median. This tells us that the data on our higher end is much much higher than our data on our lower end, which would skew our mean to be high compared to the median. The same is true of nuclear-encoded genes, but to a lower scale. 
```

------------------------------------------------------------------------

### Q5. Are liverâ€™s most highly expressed genes also highly expressed in another tissue?

Take the 20 most highly exprssed genes in **Liver** and check how they are expressed in the **Heart - Left Ventricle**. Identify:\
1. one gene that is highly expressed in both tissues\
2. one gene that is highly expressed in Liver but very low in Heart

Look up basic information about these genes on the internet. Do their expression patterns make sense given their known functions?

```{r}

dt %>%
  group_by(Liver) %>%
  arrange(desc(Liver)) %>%
  head(20)


# One gene highly expressed in both tissues is MT-ATP6
# One gene highly expressed in liver but low in heart is APOC3

#MT-ATP6 encodes one part of ATP Synthase, which is responsible for the final step of oxidative phosphorylation. This is vital to creating ATP, and therefore vital for keeping a cell alive (since it would prefer to be alive and not dead)(unless the cell is altruistic and wants to commit itself to cell death for the betterment of the body), so it would make sense for it to be highly expressed in both the liver and heart
#APOC3 encodes a component of TG-rich lipoproteins. Since the liver has functions in fat-soluble vitamin storage and metabolism, this makes sense for APOC3 to be highly expressed in the liver but not the heart
```

------------------------------------------------------------------------

### Q6. Comparing gene expression between tissues

Another way to compare tissues is to ask how similar are their expression patterns overall, across all genes?

R has a built-in function called `cor.test()` that measures the correlation between two vectors of numbers. Remember that you can extract the contents of a given column of a tibble as a vector using the syntax `tibble_name$column_name`. If a column name contains spaces, enclose the column name in backticks such as `` tibble_name$`column name` ``.

Use `cor.test()` with the argument `method = "kendall"` to compute: - the Spearman correlation between `Liver` and `Heart - Left Ventricle` - the Spearman correlation between `Liver` and `Brian - Cortex`

Which pair of tissues is more strongly correlated? Does this result make sense biologically?

```{r}

liver_tb <- dt$Liver
heartlv_tb <- dt$`Heart - Left Ventricle`
brainc_tb <- dt$`Brain - Cortex`
# the closer the tau value is to 1, the closer the correlation

# Liver vs Heart answer
cor.test(liver_tb, heartlv_tb, method = "kendall")
#The tau value is 0.8053218

# Liver vs Brain answer
cor.test(liver_tb, brainc_tb, method = "kendall")
#The tau value is 0.7342544

# Liver and Heart are more strongly correlated than the Liver and Brain are to each other. This makes sense biologically, possibly due to the fact that the liver and heart are both derived from mesoderm, whereas the brain is derived from ectoderm.

```

------------------------------------------------------------------------

### Optional: further exploration

What other patterns are present in the data? Use tidyverse functions to explore and summarize the data in different ways and see what you can learn.

```{r}
# answer here
```
